.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,184 @@
+        tofu-wrapper: [true, false]
+    defaults:
+      run:
+        working-directory: ./.github/workflows/data/local
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu
+      uses: ./
+      with:
+        tofu_wrapper: ${{ matrix['tofu-wrapper'] }}
+
+    - name: OpenTofu Init
+      run: tofu init
+
+    - name: OpenTofu Format
+      run: tofu fmt -check
+
+    - name: OpenTofu Plan
+      id: plan
+      run: tofu plan
+
+    - name: Print OpenTofu Plan
+      if: ${{ matrix['tofu-wrapper'] == 'true' }}
+      run: echo "${{ steps.plan.outputs.stdout }}"
+  tofu-credentials-cloud:
+    name: 'OpenTofu Cloud Credentials'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    env:
+      TF_CLOUD_API_TOKEN: 'XXXXXXXXXXXXXX.atlasv1.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+      - name: Setup OpenTofu
+        uses: ./
+        with:
+          cli_config_credentials_token: ${{ env.TF_CLOUD_API_TOKEN }}
+
+      - name: Validate OpenTofu Credentials (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          cat ${APPDATA}/tofu.rc | grep 'credentials "app.terraform.io"'
+          cat ${APPDATA}/tofu.rc | grep 'token = "${{ env.TF_CLOUD_API_TOKEN }}"'
+
+      - name: Validate Teraform Credentials (Linux & macOS)
+        if: runner.os != 'Windows'
+        run: |
+          cat ${HOME}/.tofurc | grep 'credentials "app.terraform.io"'
+          cat ${HOME}/.tofurc | grep 'token = "${{ env.TF_CLOUD_API_TOKEN }}"'
+
+  tofu-credentials-enterprise:
+    name: 'OpenTofu Enterprise Credentials'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    env:
+      TF_CLOUD_API_TOKEN: 'XXXXXXXXXXXXXX.atlasv1.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+      - name: Setup OpenTofu
+        uses: ./
+        with:
+          cli_config_credentials_hostname: 'tofu.example.com'
+          cli_config_credentials_token: ${{ env.TF_CLOUD_API_TOKEN }}
+
+      - name: Validate OpenTofu Credentials (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          cat ${APPDATA}/tofu.rc | grep 'credentials "tofu.example.com"'
+          cat ${APPDATA}/tofu.rc | grep 'token = "${{ env.TF_CLOUD_API_TOKEN }}"'
+
+      - name: Validate Teraform Credentials (Linux & macOS)
+        if: runner.os != 'Windows'
+        run: |
+          cat ${HOME}/.tofurc | grep 'credentials "tofu.example.com"'
+          cat ${HOME}/.tofurc | grep 'token = "${{ env.TF_CLOUD_API_TOKEN }}"'
```

Comment:
Do we need these?

Reply:
It's a good question - let's reevalute it after v0.0.1. This test was copies from setup-terraform to make sure that the action provisions cloud's configuration.